### DISCUSSION TITLE
Logging shows IP address as N/A

### DISCUSSION THREAD

**Discussion by DASCert** [kunalnagar/custom-404-pro#68] [Q&A] [answered]
Thanks for creating this plug in.

I've turned on logging for Custom 404 Pro just to see what resources bots are trying to access but the logs (and emails) do not display the User IP.  It just shows "N/A".

For example, someone/something tried to access /password/reset, apparently and the entry generated in the log is shown in the attached file.  I tried accessing the same resource (just once) and it created two entries in the log with my correct IP address.

So, I'm wondering about the circumstances under which the IP address gets reported as N/A. 

Is this a bug or as a result of some known combination of conditions?

Any ideas?

![404](https://github.com/kunalnagar/custom-404-pro/assets/37386893/e5c181b1-a050-4a5d-af90-e227d862bfb0)

**Comment by kunalnagar** [comment, marked as answer]
Hey @DASCert - so N/A means that the option for logging IP addresses was not selected when the log was recorded. Here's [the source](https://github.com/kunalnagar/custom-404-pro/blob/master/admin/AdminClass.php#L167-L168).

There is also some helper text below the option.

![image](https://github.com/kunalnagar/custom-404-pro/assets/2741371/a633adcb-444b-4092-b250-562415d79aa3)

**Comment by DASCert** [comment]
Hi and thanks for the quick response.

Unfortunately, that's not the cause.  I *do* have Log IP ticked and since it lists my IP when I try to access the same link, I know that it is working.

What else could cause the IP address to be bypassed?

![image](https://github.com/kunalnagar/custom-404-pro/assets/37386893/8d5365bb-10a1-49c4-b4d1-8a3a907ce994)

**Reply by kunalnagar** [reply]
From the code that I linked, it means that when that log was recorded, the Log IP option was not set.

**Reply by DASCert** [reply]
I understand what you're saying but I regret to tell you that the Log IP option **was** set from the start and continues to be set.  
And the problem continues to occur.

As an aside, when I try to go to a page that doesn't exist, say, www.mysite.com/bogus, two entries are placed into the log an two emails are generated.

If you have suggestions for debugging, feel free to pass on otherwise I'll put some diagnostics in and see what happens.

**Reply by kunalnagar** [reply]
Interesting. Thanks for the info. The double logging is the first time I'm hearing of the issue. Some more debugging questions for you:

- What mode have you selected i.e. WordPress page or URL?
- Have you tried switching modes and seeing if the issue is reproducible?

I'm thinking of a next step - what would be a possible way for me to repro this issue?

**Comment by DASCert** [comment]
I know why it's not working properly.
